save-backend/src/main/kotlin/org/cqfn/save/backend/service/PermissionService.kt

```diff
@@ -0,0 +1,70 @@
+package org.cqfn.save.backend.service
+
+import org.cqfn.save.backend.repository.UserRepository
+import org.cqfn.save.domain.Role
+import org.cqfn.save.entities.Project
+import org.cqfn.save.entities.User
+import org.cqfn.save.permission.SetRoleRequest
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.util.function.component1
+import reactor.kotlin.core.util.function.component2
+import reactor.util.function.Tuple2
+import java.util.Optional
+
+/**
+ * Service for managing user permissions
+ */
+@Service
+class PermissionService(
+    private val projectService: ProjectService,
+    private val userRepository: UserRepository,
+    private val lnkUserProjectService: LnkUserProjectService,
+) {
+    /**
+     * @param userName
+     * @param projectName
+     * @param organizationName
+     */
+    fun getRole(userName: String, projectName: String, organizationName: String): Mono<Role> = findUserAndProject(userName, organizationName, projectName).map { (user, project) ->
+        getRole(user, project)
+    }
+
+    /**
+     * @param user
+     * @param project
+     * @return role of [user] in [project], may throw exception if data is not consistent
+     */
+    @Suppress("UnsafeCallOnNullableType")
+    internal fun getRole(user: User, project: Project): Role = lnkUserProjectService.findRoleByUserIdAndProject(user.id!!, project)
```

Comment:
any intention to make this version only `internal`? another is `public`

Reply:
Yes, because it's used only in this service and in tests.

Reply:
Moved one of them into tests; it wasn't used anywhere else.